aggregator/src/aggregator/aggregation_job_driver/tests.rs

```diff
@@ -84,49 +84,95 @@ async fn aggregation_job_driver() {
         .to_batch_interval_start(task.time_precision())
         .unwrap();
     let batch_identifier = TimeInterval::to_batch_identifier(&leader_task, &(), &time).unwrap();
-    let report_metadata = ReportMetadata::new(random(), time);
     let verify_key: VerifyKey<VERIFY_KEY_LENGTH> = task.vdaf_verify_key().unwrap();
     let measurement = IdpfInput::from_bools(&[true]);
     let aggregation_param =
         Poplar1AggregationParam::try_from_prefixes(Vec::from([IdpfInput::from_bools(&[true])]))
             .unwrap();
+    let agg_auth_token = task.aggregator_auth_token().clone();
+    let helper_hpke_keypair = HpkeKeypair::test();
 
-    let transcript = run_vdaf(
+    let accepted_report_metadata = ReportMetadata::new(random(), time);
+    let accepted_transcript = run_vdaf(
         vdaf.as_ref(),
         verify_key.as_bytes(),
         &aggregation_param,
-        report_metadata.id(),
+        accepted_report_metadata.id(),
         &measurement,
     );
-
-    let agg_auth_token = task.aggregator_auth_token().clone();
-    let helper_hpke_keypair = HpkeKeypair::test();
-    let report = LeaderStoredReport::generate(
+    let accepted_report = LeaderStoredReport::generate(
         *task.id(),
-        report_metadata,
+        accepted_report_metadata,
         helper_hpke_keypair.config(),
         Vec::new(),
-        &transcript,
+        &accepted_transcript,
     );
 
+    let prepare_errors = [
+        PrepareError::BatchCollected,
+        PrepareError::ReportReplayed,
+        PrepareError::ReportDropped,
+        PrepareError::HpkeUnknownConfigId,
+        PrepareError::HpkeDecryptError,
+        PrepareError::VdafPrepError,
+        PrepareError::BatchSaturated,
+        PrepareError::TaskExpired,
+        PrepareError::InvalidMessage,
+        PrepareError::ReportTooEarly,
+    ];
+    let mut rejected_reports: Vec<_> = prepare_errors
+        .into_iter()
+        .map(|prepare_error| {
```

Comment:
every time I see this, I feel bad about writing `for prepare_error in ...` and being far too C.

Reply:
I certainly have a bias toward iterators, but you can only go so far with them. The language kinda nudges you toward using for loops by having some of the ceremony around borrowing elided. And more significantly, I often find myself starting out by chaining iterators, but then realizing I need to do something `async` in a closure and glumly resorting to a `for` loop. Point being: both patterns have their place. Use whatever works and don't beat yourself up.